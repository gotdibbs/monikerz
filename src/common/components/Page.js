import React from 'react';
import { Helmet } from 'react-helmet';

import Loading from './Loading';

export default function Page({ appClassName='app-body', bodyClassName = '', children, isLoading, title }) {
    return (
        <main className={appClassName}>
            <Helmet>
                <title>{title ? `Monikerz - ${title}` : 'Monikerz'}</title>
            </Helmet>
            
            <div className={bodyClassName}>
                {isLoading ? <Loading /> : children}
            </div>
        </main>
    );
}