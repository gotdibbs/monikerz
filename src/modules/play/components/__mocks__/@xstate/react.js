let state = {
    matches: jest.fn()
};

const send = jest.fn();

export const useMachine = () => {
    return [useMachine.state, useMachine.send];
};

useMachine.state = state;
useMachine.send = send;