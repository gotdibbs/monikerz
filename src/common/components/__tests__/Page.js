import React from 'react';
import { render } from '@testing-library/react';
import Page from '../Page';

jest.mock('react-helmet', () => {
    return {
        Helmet: function Helmet(props) {
            return (
                <div>{props.children}</div>
            );
        }
    };
});

jest.mock('../Loading');

describe('Page', () => {
    it('should render', () => {
        const { asFragment } = render(<Page />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should accept custom titles', () => {
        const { container, getByText } = render(<Page title='Test' />);
        expect(container).toContainElement(getByText('Monikerz - Test'));
    });

    it('should accept no title', () => {
        const { container, getByText } = render(<Page />);
        expect(container).toContainElement(getByText('Monikerz'));
    });

    it('should render a valid loading state', () => {
        const { container, queryByText, getByTestId } = render(<Page isLoading={true}>children</Page>);

        expect(container).toContainElement(getByTestId('loading'));
        expect(container).not.toContainElement(queryByText('children'));
    });

    it('should render a valid loading state', () => {
        const { container, getByText, queryByTestId } = render(<Page isLoading={false}>children</Page>);

        expect(container).not.toContainElement(queryByTestId('loading'));
        expect(container).toContainElement(getByText('children'));
    });
});