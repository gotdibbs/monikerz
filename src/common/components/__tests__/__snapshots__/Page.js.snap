// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render 1`] = `
<DocumentFragment>
  <main
    class="app-body"
  >
    <div>
      <title>
        Monikerz
      </title>
    </div>
    <div
      class=""
    />
  </main>
</DocumentFragment>
`;
